analyses/fusion-summary/01-fusion-summary.Rmd

```diff
@@ -203,9 +257,104 @@ allFuseEwing %>%
+nonkinaseLGAT <- filterFusion(df = putativeOncogenicDF,
+                             fuses = nonkinase_lgatFuses,
+                             genes = nonkinase_lgatGenes) %>%
+  distinct()
+
+# First, filter all fusion dataframe for 3' kinases which are in-frame and retain the kinase domain - keep these
+three_prime_kinase_inframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes") %>%
+  select(Sample, FusionName, Gene1A, Gene1B) %>%
+  distinct()
+
+# Which fusions are not in-frame?
+three_prime_kinase_outframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type != "in-frame") %>%
+  distinct()
+# Check they don't retain the kinase domain. They do not, so leave out.
+table(three_prime_kinase_outframe$DomainRetainedGene1B)
+
+# Filter all fusions for 5' kinase fusions which have reciprocal fusions and have lost the kinase domain. 
+five_prime_domain_lost <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1A_anno) & reciprocal_exists == "TRUE") %>%
+  select(Sample, FusionName, Gene1A, Gene1B, Fusion_Type, DomainRetainedGene1A) %>%
+  filter(DomainRetainedGene1A == "No") %>%
+  distinct()
+
+# Keep those with kinase domain retained and fusion in-frame - keep this list
+five_prime_domain_intact <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1A_anno) & reciprocal_exists == "TRUE") %>%
+  select(Sample, FusionName, Gene1A, Gene1B, Fusion_Type, DomainRetainedGene1A) %>%
+  filter(DomainRetainedGene1A == "Yes") %>%
+  distinct()
+
+# Identify reciprocal matched fusions which are in-frame
+five_prime_domain_lost$reciprocal <- paste(five_prime_domain_lost$Gene1B, five_prime_domain_lost$Gene1A, sep ="--")
+five_prime_reciprocals <- five_prime_domain_lost[,c("Sample", "reciprocal")] %>%
+  rename(FusionName = reciprocal)
+five_prime_reciprocals_full <- five_prime_reciprocals %>%
+  left_join(allFuseLGAT, by = c("Sample", "FusionName")) %>%
+  filter(Fusion_Type == "in-frame")
+# Keep the list of the original 5' kinase fusions with matched reciprocals
+five_prime_kinase_keep <- five_prime_reciprocals_full[,c("Sample", "Gene1A", "Gene1B")] %>%
+  distinct()
+  five_prime_kinase_keep$FusionName <- paste(five_prime_kinase_keep$Gene1B, five_prime_kinase_keep$Gene1A, sep ="--")
+
+# Which ones are not in-frame? These are mostly genes fused to self
+out_frame_reciprocals <- setdiff(five_prime_reciprocals[,c("Sample", "FusionName")], five_prime_reciprocals_full[,c("Sample", "FusionName")])
+out_frame_reciprocals
+
+#Check these do not retain the kinase domain. They do not, so will leave out
+out_frame_reciprocals_full <- out_frame_reciprocals %>%
+  left_join(allFuseLGAT, by = c("Sample", "FusionName")) %>%
+  select(Sample, FusionName, Gene1A, Gene1B, Fusion_Type, DomainRetainedGene1A, DomainRetainedGene1B)
+
+# Rbind lists for final table of LGAT fusions of interest
+subsetFuseLGAT <- bind_rows(nonkinaseLGAT, three_prime_kinase_inframe[,c("Sample", "FusionName")],
```

Comment:
```suggestion
subFuseLGAT <- bind_rows(nonkinaseLGAT, three_prime_kinase_inframe[,c("Sample", "FusionName")],
```
In the next chunk it seems the object `subsetFuseLGAT` should be actually named `subFuseLGAT` ?

Reply:
thanks for catching that - this is what happens when you write code at midnight! I was renaming to make things clearer, so just now, I actually updated the latter to `subsetFuseLGAT`